core/indexer/common.go

```diff
@@ -249,49 +251,57 @@ func buildReceiptTransaction(
 		SenderShard:   mb.SenderShardID,
 		GasPrice:      0,
 		GasLimit:      0,
-		Data:          rpt.Data,
+		Data:          string(rpt.Data),
 		Signature:     "",
 		Timestamp:     time.Duration(header.GetTimeStamp()),
 		Status:        "Success",
 	}
 }
 
-func serializeBulkMiniBlocks(hdrShardID uint32, bulkMbs []*Miniblock) bytes.Buffer {
-	var buff bytes.Buffer
-
+func serializeBulkMiniBlocks(hdrShardID uint32, bulkMbs []*Miniblock) (insert, update bytes.Buffer) {
 	for _, mb := range bulkMbs {
-		var err error
-		var meta, serializedData []byte
 		if hdrShardID == mb.SenderShardID {
 			//insert miniblock
-			meta = []byte(fmt.Sprintf(`{ "index" : { "_id" : "%s", "_type" : "%s" } }%s`, mb.Hash, "_doc", "\n"))
-			serializedData, err = json.Marshal(mb)
+			meta := []byte(fmt.Sprintf(`{ "index" : { "_id" : "%s", "_type" : "%s" } }%s`, mb.Hash, "_doc", "\n"))
+			serializedData, err := json.Marshal(mb)
 			if err != nil {
 				log.Debug("indexer: marshal",
 					"error", "could not serialize miniblock, will skip indexing",
 					"mb hash", mb.Hash)
 				continue
 			}
+
+			// append a newline for each element
```

Comment:
duplicate code here and in the else branch

Reply:
done.

Reply:
done.